packages/stark-ui/src/modules/toast-notification/components/toast-notification.component.ts

```diff
@@ -0,0 +1,77 @@
+import {Component, HostBinding, Inject, OnInit, ViewEncapsulation} from "@angular/core";
+import { MAT_SNACK_BAR_DATA } from "@angular/material";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkMessageType } from "../../../common";
+import { STARK_TOAST_NOTIFICATION_SERVICE, StarkToastNotificationService } from "../services";
+import { StarkToastNotificationMessage } from "../components";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-toast-notification";
+
+/**
+ * Component display stark's toast notification (based on angular material's snackbar) with custom html
+ */
+@Component({
+	selector: "stark-toast-notification",
+	templateUrl: "./toast-notification.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkToastNotificationComponent implements OnInit {
+	/**
+	 * Adds class="stark-toast-notification" attribute on the host component
+	 */
+	@HostBinding("class")
+	public class: string = componentName;
+	
+	/**
+	 * The message data linked to the toast notification.
+	 */
+	public message: StarkToastNotificationMessage;
+
+	/**
+	 * Class constructor
+	 * @param logger - The logger of the application
+	 * @param toastService - The toast notification service
+	 * @param data - the data linked to the toast notification
+	 */
+	public constructor(
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		@Inject(STARK_TOAST_NOTIFICATION_SERVICE) public toastService: StarkToastNotificationService,
+		@Inject(MAT_SNACK_BAR_DATA) public data: StarkToastNotificationMessage
+	) {
+		this.message = data;
+		this.logger.debug(componentName + ": data received : %o", this.message);
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnInit(): void {
+		this.logger.debug(componentName + ": controller initialized");
+	}
+	
+	/**
+	 * Closes the toast
+	 */
+	public closeToast(): void {
+		this.toastService.hide();
+	}
+
+	/**
+	 * Generate the css class of the toast notification based on its type
+	 * @returns a string containing the css class of the toast notification
+	 */
+	public getMessageTypeClass(): string {
+		switch (this.message.type) {
+			case StarkMessageType.WARNING:
+				return "stark-toast-message-warning";
+			case StarkMessageType.ERROR:
+				return "stark-toast-message-error";
+			case StarkMessageType.INFO:
+			default:
+				return "stark-toast-message-info";
```

Comment:
Could you change the default case to log an error that the message type is unknown (like in the old Stark)?

We should be strict on this to prevent the developers from misusing our API, so if an error is logged then it means they are using it wrong ;)